lib/daru/vector.rb

```diff
@@ -170,21 +171,26 @@ def map!(&block)
     #   vecarr = Daru::Vector.new [1,2,3,4], index: [:a, :e, :i, :o]
     #   vechsh = Daru::Vector.new({a: 1, e: 2, i: 3, o: 4})
     def initialize source, opts={}
-      index, source = parse_source(source, opts)
-      set_name opts[:name]
+      if opts[:type] == :category
+        extend Daru::Category
```

Comment:
For me, this part of code and overall logic looks like we just want a `CategoryVector` class, not some playing with module-inclusion-on-the-fly.